Resolves #1264: remove undo redo buttons

aileenzeng opened the pull request:
For #1264

Changes made:
* Updated UI - no longer contains undo/redo buttons
* Removed all instances of ActionStack from the code. (ActionStack seemed to be responsible for all undo/redo functionality)

| Before | After |
| ------ | ------ |
| ![image](https://user-images.githubusercontent.com/25534091/42190237-d4d13178-7e10-11e8-8a0e-d73c98b26afe.png) | ![image](https://user-images.githubusercontent.com/25534091/42190240-da898480-7e10-11e8-8163-4d0336ebc0b3.png) |

misaugstad commented:
Wow, I can't wait to review this PR!! Deleting 2000 lines of code brings me so much joy :heart_eyes:

misaugstad commented:
Works pretty well! The only thing that needs fixing is the locations of the messages in the tutorial that instructs users to click the zoom in and out buttons :) It seems that we both missed this during testing when adding the NoSidewalk label though!

![tutorial_zoom_in_message_placement](https://user-images.githubusercontent.com/6518824/42287475-5944b546-7f6b-11e8-95a6-b731dee50ccd.png)

misaugstad commented:
Also the zoom buttons are a bit smaller in the current implementation than in your mockup.

Mockup:
![image](https://user-images.githubusercontent.com/25534091/41754745-8ea2c86e-7588-11e8-806a-f2694ecdfb34.png)

Current implementation:
![zoom_buttons_smaller](https://user-images.githubusercontent.com/6518824/42288646-71acff30-7f70-11e8-92fb-2ecb1c663124.png)

I feel like the buttons might look a bit better being slightly larger. Did you choose the smaller buttons for a reason, or had you meant to make them larger? I trust your design decisions here, just bringing up that the sizing is different from the mockups you made :)

aileenzeng commented:
I'll look more into both issues! For the buttons part, I think I was copying some of the old CSS (whereas in the mockups I more-or-less eyeballed it).

aileenzeng commented:
I fixed the positions of the zoom buttons in Onboarding:

| Zoom In | Zoom Out |
| -------- | ---------- |
| ![image](https://user-images.githubusercontent.com/25534091/42339277-64e71f2a-8041-11e8-8235-e92bf30c306f.png) | ![image](https://user-images.githubusercontent.com/25534091/42339283-69dfd738-8041-11e8-8f6a-074eb3c0ab30.png) |

The zoom buttons should look more like the mockup on my computer - I modified the height by 1px - nothing major). I'm not super sure why they had a thick border earlier on (I couldn't replicate that issue)
![image](https://user-images.githubusercontent.com/25534091/42339335-8ccabb00-8041-11e8-86da-febd9e993d63.png)

misaugstad commented:
Nice fix!

Okay the only other thing I could find is that the tooltip that shows what the keyboard shortcut is for the zoom out now has a line break that doesn't look good.

Old look:
![zoom-out-shortcut-good](https://user-images.githubusercontent.com/6518824/42391112-164c5174-8103-11e8-8e43-609fd6f7a7c5.png)

Current look:
![zoom-out-shortcut-bad](https://user-images.githubusercontent.com/6518824/42391122-20d78fe6-8103-11e8-8e00-b0786294fdf5.png)

Could you just make the tooltip show in one line? Then it should be ready for code review :)

aileenzeng commented:
Oops, good catch!

Here's the updated version:
![image](https://user-images.githubusercontent.com/25534091/42400864-0d5b252a-8128-11e8-9845-0cef3dfadfa1.png)

misaugstad commented:
The tooltip still had a linebreak on it when I tested, so I increased the width a little bit more in the same way that you had. Here is the before and after:
![zoom-out-170](https://user-images.githubusercontent.com/6518824/42403052-76353c42-8133-11e8-9330-b2fb0a8b65ed.png)
![zoom-out-185](https://user-images.githubusercontent.com/6518824/42403054-77a3fe9c-8133-11e8-8a53-b26a67582608.png)


Could you just test it to see if that looks okay in your browser as well? I'll start reviewing the code now :)

misaugstad commented on public/javascripts/SVLabel/doc/PointCloud.html:
```diff
@@ -583,7 +583,11 @@ <h4 class="name" id="~search"><span class="type-signature">(inner) </span>search

ix = idx / 3 % w
iy = (idx / 3 - ix) / w
+    Given the coordinate x, y (and z), return index of the point cloud data.
```
What happened with all these doc files? I understand how the ones with action stack references were changed, but what happened with the line breaks in these ones?

aileenzeng commented:
The zoom tooltip looks good to me! 

I don't remember changing the PointCloud doc file - I could revert that change to it's previous state, since I'm not super sure what it does.

misaugstad commented:
Upon further web searching, it seems that the ^M characters are actually a byproduct of the newlines being created on a Windows system [https://en.wikipedia.org/wiki/Newline#Issues_with_different_newline_formats](https://en.wikipedia.org/wiki/Newline#Issues_with_different_newline_formats). So you actually didn't change anything, the newlines are just being represented differently :)

Thus, this PR is ready for merging! Wooooo nice work!!
